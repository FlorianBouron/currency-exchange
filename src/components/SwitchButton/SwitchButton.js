import React from "react";
import PropTypes from "prop-types";
import Fab from "@material-ui/core/Fab";
import CompareArrows from "@material-ui/icons/CompareArrows";
import styles from "./SwitchButton.module.scss";

const SwitchButton = ({ className, onClick }) => {
  return (
    <Fab
      aria-label="Switch"
      className={[styles["switch-button-container"], className].join(" ")}
      size="small"
      onClick={onClick}
    >
      <CompareArrows className={styles["switch-button__icon"]} />
    </Fab>
  );
};

SwitchButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

SwitchButton.defaultProps = {
  className: null
};

export default SwitchButton;
