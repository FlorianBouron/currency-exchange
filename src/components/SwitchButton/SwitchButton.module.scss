@import "../../constants/colors.scss";

.switch-button-container {
  background-color: $white !important;
  color: $primary-blue !important;
  .switch-button__icon {
    transform: rotate(90deg);
  }
}
