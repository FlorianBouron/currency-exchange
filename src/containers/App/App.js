import React from "react";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import CurrentRate from "../../components/CurrentRate";
import styles from "./App.module.scss";

const App = () => {
  return (
    <div className={styles["app-container"]}>
      <Paper className={styles["app-paper"]}>
        <Typography variant="h5" component="h3" className={styles["app-title"]}>
          Exchange
        </Typography>
        <CurrentRate rateFrom="1zl" rateTo="0,231â‚¬" />
        <Button variant="contained" color="secondary">
          Exchange
        </Button>
      </Paper>
    </div>
  );
};

export default App;
