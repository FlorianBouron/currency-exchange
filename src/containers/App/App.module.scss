@import "../../constants/utils.scss";
@import "../../constants/colors.scss";

.app-container {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;

  .app__paper {
    width: px2rem(400px);
    padding: px2rem(20px);
  }

  .app__title {
    text-align: center;
  }

  .app__switch-button {
    position: absolute;
    top: calc(50% - #{px2rem(14px)});
    transform: translate(-50%, -50%);
    margin-left: 20px;
  }

  .app__current-rate {
    position: absolute;
    top: calc(50% - #{px2rem(14px)});
    left: 50%;
    transform: translate(-50%, -50%);
  }

  .app__currency-container-read-only {
    margin: 0 px2rem(-20px) px2rem(20px) px2rem(-20px);
    padding: px2rem(30px) px2rem(40px);
    background-color: $primary-gray;
  }

  .app__footer {
    display: flex;
    justify-content: center;
  }
}
