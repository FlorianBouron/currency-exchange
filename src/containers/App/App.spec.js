import React from "React";
import { shallow, mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import Button from "@material-ui/core/Button";
import SwitchButton from "../../components/SwitchButton";
import App from ".";

describe("App component", () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const reduxState = {
    currencies: {
      currencyFrom: { name: "GBP", symbol: "£", inputValue: "" },
      currencyTo: { name: "EUR", symbol: "€", inputValue: "" }
    },
    wallets: {
      data: [
        { id: "wallet-id-0", name: "GBP", amount: 200, symbol: "£" },
        { id: "wallet-id-1", name: "EUR", amount: 85, symbol: "€" },
        { id: "wallet-id-2", name: "USD", amount: 1000, symbol: "$" }
      ]
    },
    rates: {
      base: "GBP",
      rates: { USD: 1.2721606179, EUR: 1.1292034599, GBP: 1 },
      error: ""
    },
    errors: { errorBalanceFrom: "" }
  };
  const store = mockStore(reduxState);

  const basicProps = {
    store
  };

  it("has working component", () => {
    const component = shallow(<App {...basicProps} />);
    expect(component).toBeDefined();
  });

  it("expects component to never be changed", () => {
    const component = shallow(<App {...basicProps} />);
    expect(component).toMatchSnapshot();
  });

  it("expects to call handleClickSwitch", () => {
    const component = mount(
      <Provider {...basicProps}>
        <App />
      </Provider>
    );
    component
      .find(SwitchButton)
      .first()
      .simulate("click");
    expect(component).toBeDefined();
  });

  it("expects to call handleClickExchange", () => {
    const component = mount(
      <Provider {...basicProps}>
        <App />
      </Provider>
    );
    component
      .find(Button)
      .first()
      .simulate("click");
    expect(component).toBeDefined();
  });
});
