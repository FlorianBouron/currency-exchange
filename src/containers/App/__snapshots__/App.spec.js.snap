// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component expects component to never be changed 1`] = `
<ContextProvider
  value={null}
>
  <App
    currencies={
      Object {
        "currencyFrom": Object {
          "inputValue": "",
          "name": "GBP",
          "symbol": "£",
        },
        "currencyTo": Object {
          "inputValue": "",
          "name": "EUR",
          "symbol": "€",
        },
      }
    }
    errorRate=""
    exchangeCurrency={[Function]}
    fetchRates={[Function]}
    isNoBalanceError={false}
    rates={
      Object {
        "EUR": 1.1292034599,
        "GBP": 1,
        "USD": 1.2721606179,
      }
    }
    setCurrencies={[Function]}
    setCurrencyFrom={[Function]}
    setCurrencyTo={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    wallets={
      Array [
        Object {
          "amount": 200,
          "id": "wallet-id-0",
          "name": "GBP",
          "symbol": "£",
        },
        Object {
          "amount": 85,
          "id": "wallet-id-1",
          "name": "EUR",
          "symbol": "€",
        },
        Object {
          "amount": 1000,
          "id": "wallet-id-2",
          "name": "USD",
          "symbol": "$",
        },
      ]
    }
  />
</ContextProvider>
`;
